search_service/proxy/atlas.py

```diff
@@ -0,0 +1,230 @@
+        """
+        Construct Atlas client that connects to cluster at {host} and {port}
+        and authenticates using {user} and {password}
+        :return: Atlas client object
+        """
+        return Atlas(host, username=user, password=password)
+
+    @staticmethod
+    def _entities(collections: EntityCollection) -> List[Entity]:
+        """
+        Helper method for flattening all collections from {collections}
+        :return: list of all entities
+        """
+        entities: List[Entity] = []
+        for collection in collections:
+            entities.extend(collection.entities)
+        return entities
+
+    def _parse_results(self, response: EntityCollection) -> List[Table]:
+        """
+        based on an atlas {response} with table entities, we map the required information
+        :return: list of tables
+        """
+        table_results = []
+        ids = list()
+        for hit in response:
+            ids.append(hit.guid)
+        # receive all entities
+        entities = self._entities(self.atlas.entity_bulk(guid=ids))
+        db_ids = []
+        for entity in entities:
+            relations = entity.relationshipAttributes
+            database = relations.get(self.DB_ATTRIBUTE)
+            if database:
+                db_ids.append(database['guid'])
+
+        # request databases
+        dbs_list = self._entities(self.atlas.entity_bulk(guid=db_ids)) if len(db_ids) > 0 else []
+        dbs_dict: Dict[str, Entity] = {db.guid: db for db in dbs_list}
+        for entity in entities:
+            relations = entity.relationshipAttributes
+            attrs = entity.attributes
+            database = relations.get(self.DB_ATTRIBUTE)
+            if database and database['guid'] in dbs_dict:
+                db_entity = dbs_dict[database['guid']]
+                db_attrs = db_entity.attributes
+                db_name = db_attrs.get(self.NAME_ATTRIBUTE, '')
+                db_cluster = db_attrs.get('clusterName', '')
+            else:
+                db_cluster = ''
+                db_name = ''
+
+            tags = []
+            # Using or in case, if the key 'classifications' is there with attrs None
+            for classification in attrs.get("classifications") or list():
+                tags.append(
+                    classification.get('typeName')
+                )
+
+            # TODO: Implement columns
+            columns: List[str] = []
+            # for column in attrs.get('columns') or list():
+            #     col_entity = entity.referredEntities[column['guid']]
+            #     col_attrs = col_entity['attributes']
+            #     columns.append(col_attrs.get(self.NAME_KEY))
+
+            table = Table(name=attrs.get(self.NAME_ATTRIBUTE),
+                          key=attrs.get(self.NAME_ATTRIBUTE),
+                          description=attrs['description'],
+                          cluster=db_cluster,
+                          database=db_name,
```

Comment:
I think this would be entityType. like the db_engine? i.e., hive, sql, postgres etc.

Reply:
Correctly fixed this, as in metaservice